patches/minecraft/net/minecraft/data/Main.java.patch

```diff
@@ -26,7 +28,7 @@
 +         Collection<Path> existingPacks = optionset.valuesOf(existing).stream().map(Paths::get).collect(Collectors.toList());
 +         java.util.Set<String> mods = new java.util.HashSet<>(optionset.valuesOf(mod));
 +         boolean isFlat = mods.isEmpty() || optionset.has(flat);
-+         net.minecraftforge.fml.DatagenModLoader.begin(mods, path, inputs, existingPacks, flag2, flag1, flag3, flag4, flag5, isFlat);
++         net.minecraftforge.fml.DatagenModLoader.begin(mods, path, inputs, existingPacks, optionset.valueOf(assetIndex), optionset.valueOf(assetsDir).toPath(), flag2, flag1, flag3, flag4, flag5, isFlat);
```

Comment:
This will NPE if someone doesn't specify the assetsDir, its not marked as required, and it shouldn't be. So you can't do .toPath directly. You need a null check